sky/clouds/gcp.py

```diff
@@ -142,6 +215,12 @@ class GCP(clouds.Cloud):
         'https://skypilot.readthedocs.io/en/latest/getting-started/installation.html#google-cloud-platform-gcp'  # pylint: disable=line-too-long
     )
 
+    def __init__(self):
+        super().__init__()
+
+        self._list_reservations_cache = cachetools.TLRUCache(
+            maxsize=1, ttu=_list_reservations_cache_ttu, timer=datetime.now)
```

Comment:
Can we use the `TTLCache` instead? It seems the `TLRUCache` is not included in the older `cachetools`, and `ray` restricts the version to be `<5.0`.
```suggestion
        self._list_reservations_cache = cachetools.TTLCache(
            maxsize=1, ttl=datetime.timedelta(300), timer=datetime.now)
```